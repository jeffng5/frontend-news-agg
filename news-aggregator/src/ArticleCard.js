import React from 'react'
import "./css/FrontPage.css"
import Archive from "./Archive"
import ColoredLineThin from "./ColoredLineThin"
import ButtonForum from "./ButtonForum"


let username = localStorage.getItem('username')
console.log(username)
// Card template to hold the articles returned

//receiving prop data from parent
const ArticleCard = ({title, description, url, author, urlToImage}) => {

  
    return (
    <>
  
    <div className='article'>
    <p><img className = 'photograph' src={urlToImage} alt = 'news'/></p>
    <h2 className='headline' id='headline'><span><a href={url} target="_blank">{title}<h5 className='desc'>{description}</h5></a></span></h2>
    </div>
    <div className='button-6'>
    <Archive username={username} title={title} description={description} url={url} author={author}/>
    </div>
    <div className='button-6'>
    <ButtonForum username={username} title={title} description={description} url={url} author={author} urltoImage={urlToImage}/></div>
    <div className='author-2'>by: {author}</div>
    <div className='line'>
    <ColoredLineThin color = "gray" />  
    </div>
   
    </>
)

}
    


export default ArticleCard;