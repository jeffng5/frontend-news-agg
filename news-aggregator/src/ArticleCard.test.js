import React from 'react';
import { render } from '@testing-library/react';
import ArticleCard from './ArticleCard'

it('renders without crashing', function() {
  render(<ArticleCard />);
})

//snapshot test
it("matches snapshot", function() {
    const {asFragment} = render(<ArticleCard />);
    expect(asFragment()).toMatchSnapshot();
  });

  test('testing component', ()=> {
    const {getByText} = render( <ArticleCard />);
    getByText('by:')
})