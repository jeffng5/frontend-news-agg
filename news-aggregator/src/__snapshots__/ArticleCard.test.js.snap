// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="article"
  >
    <p>
      <img
        alt="news"
        class="photograph"
      />
    </p>
    <h2
      class="headline"
      id="headline"
    >
      <span>
        <a
          target="_blank"
        >
          <h5
            class="desc"
          />
        </a>
      </span>
    </h2>
  </div>
  <div
    class="button-6"
  >
    <button
      class="archive"
      style="background-color: yellow;"
    >
      <img
        alt=""
        id="archive"
        src="penfeather.png"
      />
      Archive
    </button>
  </div>
  <div
    class="button-6"
  >
    <button
      class="forum"
      style="background-color: aquamarine;"
    >
      <h6
        id="button"
      >
        Post to Forum
      </h6>
    </button>
  </div>
  <div
    class="author-2"
  >
    by: 
  </div>
  <div
    class="line"
  >
    <hr
      style="color: gray; background-color: gray; height: 3px;"
    />
  </div>
</DocumentFragment>
`;
